python/ptest/cases/gomspace_checkout_case.py

```diff
@@ -0,0 +1,126 @@
+            simulation.flight_controller.write_state(string_state, state_value)
+            return read_state(self, string_state)
+
+        # readable fields
+
+        vboost = [int(read_state(self, "gomspace.vboost.output" + str(i)))
+                  for i in range(1, 4)]
+        for n in range(0, len(vboost)):
+            print("vboost" + str(n) + " is: " + str(vboost[n]) + " mV")
+
+        vbatt = int(read_state(self, "gomspace.vbatt"))
+        if vbatt < 6000 or vbatt > 8400:
+            print(
+                "Vbatt is out of expected range [6000, 8400] mV at: " + str(vbatt))
+
+        curin = [int(read_state(self, "gomspace.curin.output" + str(i)))
+                 for i in range(1, 4)]
+        for n in range(0, len(curin)):
+            print("curin" + str(n) + " is: " + str(curin[n]) + " mA")
+        cursun = int(read_state(self, "gomspace.cursun"))
+        print("cursun is: " + str(cursun) + " mA")
+        cursys = int(read_state(self, "gomspace.cursys"))
+        print("cursys is: " + str(cursys) + " mA")
+
+        curout = [int(read_state(self, "gomspace.curout.output" + str(i)))
+                  for i in range(1, 7)]
+        for n in range(0, len(curout)):
+            print("curout" + str(n) + " is: " + str(curout[n]) + " mA")
+
+        output = [bool(read_state(self, "gomspace.output.output" + str(i)))
+                  for i in range(1, 7)]
+
+        for n in range(0, len(output)):
+            out_n = output[n]
+            # checked umbilical for which outputs should be 0 and 1:
+            # 1-5 are all 5V, 6 is 3.3V
+            if out_n is False:
+                print("Output-" + n +
+                      " is not on: ")
+
+        wdt_i2c_time_left = int(read_state(self, "gomspace.wdt_i2c_time_left"))
+        if wdt_i2c_time_left < 99:
+            print("wdt_i2c_time_left is less than 99 seconds at: " +
+                  str(wdt_i2c_time_left))
+
+        counter_wdt_i2c = read_state(self, "gomspace.counter_wdt_i2c")
+        print("counter_wdt_i2c is: " + str(counter_wdt_i2c))
+        counter_boot = read_state(self, "gomspace.counter_boot")
+        print("counter_wdt_i2c is: " + str(counter_boot))
+
+        temp = [int(read_state(self, "gomspace.temp.output" + str(i)))
+                for i in range(1, 5)]
+        for n in range(0, len(temp)):
+            temp_n = temp[n]
+            if temp_n < 20 or out_n > 22:
+                print("Temp-" + str(n) +
+                      " is out of room temperature range [20, 25] degC at: " + str(temp_n))
+
+        bootcause = read_state(self, "gomspace.bootcause")
+        print("bootcause is: " + str(bootcause))
+        battmode = read_state(self, "gomspace.battmode")
+        print("battmode is: " + str(battmode))
+        pptmode = read_state(self, "gomspace.pptmode")
+        print("pptmode is: " + str(pptmode))
+
+        # writable fields
+        power_cycle_output_cmd = [read_state(self, "gomspace.power_cycle_output" + str(i) + "_cmd")
```

Comment:
By default the power cycle commands should be false. You can set them to true to start a power cycle. Once the power cycle is over, the command statefields should automatically be false again